lib/elasticsearch/elastic-query-builder.js

```diff
@@ -464,6 +457,37 @@ class ElasticQueryBuilder {
     this.query.addMust(should.length === 1 ? should[0] : { bool: { should } })
   }
 
+  /**
+  * Handle use of subject=, contributor=, title=, & callnumber= Adv Search
+  * params.
+  * Note that the RC Adv Search page may also apply filters, which are
+  * handled by `applyFilters`.
+  **/
+  applyAdvancedSearchParams () {
+    // We're specifically interested in params that match supported search-
+    // scopes because we can build a whole ES query just using the existing
+    // logic for that search scope:
+    if (this.request.advancedSearchParamsThatAreAlsoScopes()) {
+      this.request
+        .advancedSearchParamsThatAreAlsoScopes()
+        .forEach((advSearchParam) => {
+          const advSearchValue = this.request.params[advSearchParam]
+          // Build a new ApiRequest object for this search-scope:
+          const request = ApiRequest.fromParams({
```

Comment:
is this pattern being used elsewhere? something about it makes me uneasy...

Reply:
Because `ApiRequest` feels like something that should only ever wrap a real user-initiated API request? It's a light-weight model representing a user query. The Adv Search page essentially allows you to find the intersection of multiple queries, so I think it's fine, but can you say more about what is unsettling? Conceptually we do want the same query to be generated from `q=foo&search_scope=title` as is generated from `title=foo`, so the two routes need to ultimately share some code. Do you see a clearer way?